File: project/backend/src/main/java/co/edu/utp/isc/gia/examsapp/web/dto/AnswerOptionDto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.edu.utp.isc.gia.examsapp.web.dto;

import co.edu.utp.isc.gia.examsapp.data.entity.Question;
import co.edu.utp.isc.gia.examsapp.data.entity.SelectedResponse;
import java.io.Serializable;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 *
 * @author CJ
 */
@AllArgsConstructor @NoArgsConstructor
@Setter @Getter
public class AnswerOptionDto implements Serializable {
    
    private Long id;
    private String index;
    private String description;
    private Boolean correctAnswer;
    private Double weight;
    private List<SelectedResponse> selectedResponses;
    private Question question;
}

File: project/frontend/src/app/dto/AnswerOptionDto.ts
import {OnInit } from '@angular/core';
import { QuestionDto } from './abstractDto/QuestionDto';
import { SelectedResponseDto } from './SelectedResponseDto';
import { OpenQuestionDto } from './OpenQuestionDto';

export class AnswerOptionDto implements OnInit {
    public id : number;
    public index : String;
    public description : String;
    public correctAnswer : Boolean;
    public weight : number;
    public selectedResponses : Array<SelectedResponseDto>;
    public question : QuestionDto;

    ngOnInit() {
        this.id = 0;
        this.index = new String;
        this.description = new String;
        this.correctAnswer = new Boolean;
        this.weight = 0.0;
        this.selectedResponses = new Array<SelectedResponseDto>();
        this.question = null;
    }

}
